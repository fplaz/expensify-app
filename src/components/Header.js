import React from 'react'
import {NavLink} from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Expensify App</h1>
        <NavLink to="/" className={({isActive}) => isActive ? "is-active" : undefined }>Home</NavLink>
        <NavLink to="/create" className={({isActive}) => isActive ? "is-active" : undefined }>Create expense</NavLink>
        <NavLink to="/help" className={({isActive}) => isActive ? "is-active" : undefined }>Help expense</NavLink>
    </header>
)

export default Header