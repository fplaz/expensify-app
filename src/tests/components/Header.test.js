import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import {render} from '@testing-library/react'
import Header from '../../components/Header'
//import ReactShallowRenderer from 'react-test-renderer/shallow'

test('Should render Header correctly', () => {
    const {asFragment} = render(
        <MemoryRouter>
            <Header />
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
    
    // const renderer = new ReactShallowRenderer()
    // renderer.render(<Header />)
    // expect(renderer.getRenderOutput()).toMatchSnapshot()
})