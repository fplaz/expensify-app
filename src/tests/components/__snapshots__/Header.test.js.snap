// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<DocumentFragment>
  <header>
    <h1>
      Expensify App
    </h1>
    <a
      aria-current="page"
      class="is-active"
      href="/"
    >
      Home
    </a>
    <a
      href="/create"
    >
      Create expense
    </a>
    <a
      href="/help"
    >
      Help expense
    </a>
  </header>
</DocumentFragment>
`;
